browser/widevine/widevine_utils.cc

```diff
@@ -5,79 +5,72 @@
 
 #include "brave/browser/widevine/widevine_utils.h"
 
+#include "base/files/file_util.h"
+#include "base/path_service.h"
+#include "base/task/task_traits.h"
+#include "base/task/thread_pool.h"
 #include "brave/browser/brave_browser_process_impl.h"
 #include "brave/browser/widevine/widevine_permission_request.h"
 #include "brave/common/pref_names.h"
 #include "brave/grit/brave_generated_resources.h"
+#include "chrome/browser/component_updater/widevine_cdm_component_installer.h"
 #include "chrome/browser/profiles/profile.h"
+#include "chrome/common/chrome_paths.h"
 #include "components/permissions/permission_request_manager.h"
 #include "components/prefs/pref_registry_simple.h"
 #include "components/prefs/pref_service.h"
 #include "components/pref_registry/pref_registry_syncable.h"
 #include "content/public/browser/browser_thread.h"
-
-#if BUILDFLAG(BUNDLE_WIDEVINE_CDM)
-#include <string>
-
-#include "base/bind.h"
-#include "brave/browser/brave_drm_tab_helper.h"
-#include "brave/browser/widevine/brave_widevine_bundle_manager.h"
-#include "chrome/browser/lifetime/application_lifetime.h"
-#include "chrome/browser/ui/browser_finder.h"
-#include "chrome/browser/ui/tabs/tab_strip_model.h"
-#endif
-
-#if BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
-#include "chrome/browser/component_updater/widevine_cdm_component_installer.h"
-#endif
+#include "third_party/widevine/cdm/widevine_cdm_common.h"
 
 using content::BrowserThread;
 
 namespace {
 
-void ClearWidevinePrefs(Profile* profile) {
-  PrefService* prefs = profile->GetPrefs();
-  prefs->ClearPref(kWidevineOptedIn);
+#if defined(OS_LINUX)
+constexpr char kWidevineInvalidVersion[] = "";
 
-#if BUILDFLAG(BUNDLE_WIDEVINE_CDM)
-  prefs->ClearPref(kWidevineInstalledVersion);
-#endif
-}
+// Added 11/2020.
+constexpr char kWidevineInstalledVersion[] = "brave.widevine_installed_version";
 
-#if BUILDFLAG(BUNDLE_WIDEVINE_CDM)
-content::WebContents* GetActiveWebContents() {
-  if (Browser* browser = chrome::FindLastActive())
-    return browser->tab_strip_model()->GetActiveWebContents();
-  return nullptr;
-}
-
-bool IsActiveTabRequestedWidevine() {
-  bool is_active_tab_requested_widevine = false;
-  if (content::WebContents* active_web_contents = GetActiveWebContents()) {
-    BraveDrmTabHelper* drm_helper =
-        BraveDrmTabHelper::FromWebContents(active_web_contents);
-    is_active_tab_requested_widevine = drm_helper->ShouldShowWidevineOptIn();
+void OnDeletedOldWidevineBinary(bool result) {
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
+  if (result) {
+    auto* local_state = g_browser_process->local_state();
+    local_state->ClearPref(kWidevineInstalledVersion);
   }
-
-  return is_active_tab_requested_widevine;
 }
 
-void OnWidevineInstallDone(const std::string& error) {
-  if (!error.empty()) {
-    LOG(ERROR) << __func__ << ": " << error;
-    return;
-  }
+bool DoDeleteOldWidevineBinary() {
+  base::FilePath widevine_base_path;
+  CHECK(base::PathService::Get(chrome::DIR_USER_DATA,
```

Comment:
I'm slightly concerned with these `CHECK`s, since `CHECK` is for critical situation when the browser just cannot work anymore (i.e. detecting a security breach). Inability to delete the widevine binary (or get a path) doesn't look like a critical problem that prevents the program from working